### DISCUSSION TITLE
Is there a way to download books that don’t have a download acsm or epub option?

### DISCUSSION THREAD

**Discussion by jasperhooloop** [sgmoore/libby-calibre-plugin#4] [Q&A]
I know there was a way before the overdrive option was removed. I'm so new to GitHub so if this is the wrong place to post this oh boy please tell me where is lol. You guys are great and calibre is amazing, thank you so much. 
So I use Libby with calibre and download my books and convert them to pdf. But some of them there is no download option, only an option to open them in browser. I know how to save the HAR files and then find the link for audiobooks, is there a similar method for books that only open in browser on Libby?
Again thank you guys so much. I added a screenshot and blacked out the book in case it's like a copyright issue, if you guys need/want to know which book I can say which one lol. There's a few in this series that don't have a download option. ![image](https://github.com/user-attachments/assets/99367381-0b4b-40ad-bb08-f9f63b51b9f4)

**Comment by sgmoore** [comment]
> Is there a way to download books that don’t have a download acsm or epub option?

Short answer is : not using this plugin and I'm not aware of any automatic way.

> I'm so new to GitHub so if this is the wrong place to post this oh boy please tell me where is lol.

It is not "wrong", but may not be the best place because this plugin doesn't even **try** to work around these issues. However I'm not sure where the best place is, but if you are using a project to automatic parsing and download the HAR files for audiobooks, perhaps there would be people there who have tried to do the same for books.

Since overdrive removed the options to do automatic downloads this plugin does not automatically download any loans but simply re-directs you to the libby webpage which may or may not allow you to download the book.

But even before the changes, there was always certain items which could not be downloaded and overdrive said that this was because the book was not available in a downloadable format, i.e. it was html only or just a series of jpegs.

When overdrive stopped allowing magazines to be downloaded, I seen one attempt to download the html pages and stitch them together to make an epub, but the results did not impress me. 

> I added a screenshot and blacked out the book in case it's like a copyright issue,

I don't think it will help knowing which book it is, but there should be no problem sharing links to books as libby provide a way to do. 
In this plugin, just right-click on a book (on the loans, hold or search tabs) and it brings up a menu with a "Copy Libby Share links" which copies the link into your clipboard , or you can do it from the website (see https://help.libbyapp.com/en-us/6210.htm?TocPath=Home%7CFrequently%20asked%20questions%7C_____8) and probably the app. 

Sorry I can't help.
